pkg/controller/dynamicupgradeableconditionscontroller.go

```diff
@@ -0,0 +1,245 @@
+		return nil, err
+	}
+
+	var incompatibleOperators []string
+
+	ceList, err := c.clusterExtensionClient.Informer().Lister().List(labels.NewSelector())
+	if err != nil {
+		c.logger.V(4).Error(err, "Error listing cluster extensions")
+		return nil, err
+	}
+
+	store := c.buildHelmStore(c.kubeclient.CoreV1().Secrets("openshift-operator-controller"))
+
+	// Get all ClusterExtensions incompatible with next Y-stream
+	for _, obj := range ceList {
+		ce, ok := obj.(*unstructured.Unstructured)
+		if !ok {
+			c.logger.V(4).Error(nil, fmt.Sprintf("Unable to Cast Object to unstructured.Unstructured. Object is of type %T", obj))
+			return nil, nil
+		}
+
+		rel, err := store.Deployed(ce.GetName())
+		if errors.Is(err, driver.ErrNoDeployedReleases) {
+			c.logger.Info("Cluster Extension not yet deployed - will check again later")
+			continue
+		}
+		if err != nil {
+			c.logger.V(4).Error(err, "Error returning the last deployed release")
+			return nil, err
+		}
+		props, err := propertyListFromPropertiesAnnotation(rel.Chart.Metadata.Annotations["olm.properties"])
+		if err != nil {
+			c.logger.V(4).Error(err, "Error converting olm.properties")
+			return nil, err
+		}
+		for _, p := range props {
+			if p.Type == MaxOpenShiftVersionProperty {
+				version := string(p.Value)
+				maxOCPVersion, err := toSemver(version)
+				if err != nil {
+					c.logger.V(4).Info(fmt.Sprintf("Error converting to semver for version %s: %v", version, err))
+					continue
+				}
+				if maxOCPVersion != nil && !maxOCPVersion.GTE(next) {
+					// Incompatible
+					incompatibleOperators = append(incompatibleOperators, rel.Labels[BundleNameKey])
+				}
+			}
+		}
+	}
+
+	// deterministic ordering
+	sort.Strings(incompatibleOperators)
+
+	updateStatusFuncs := make([]v1helpers.UpdateStatusFunc, 0, len(incompatibleOperators))
+	if len(incompatibleOperators) > 0 {
+		updateStatusFuncs = append(updateStatusFuncs, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "Upgradeable",
+			Status:  operatorv1.ConditionFalse,
+			Reason:  incompatibleOperatorsInstalled,
+			Message: strings.Join(incompatibleOperators, ","),
+		}))
+	} else {
+		updateStatusFuncs = append(updateStatusFuncs, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "Upgradeable",
+			Status: operatorv1.ConditionTrue,
+		}))
+	}
+	return updateStatusFuncs, nil
+}
+
+func propertyListFromPropertiesAnnotation(raw string) ([]property.Property, error) {
+	var props []property.Property
+	if err := json.Unmarshal([]byte(raw), &props); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal properties annotation: %w", err)
+	}
+	return props, nil
+}
+
+func (c *dynamicUpgradeableConditionController) buildHelmStore(secretClient v1.SecretInterface) helm.Storage {
+	log := func(s string, args ...interface{}) { c.logger.V(4).Info(fmt.Sprintf(s, args...)) }
+	csConfig := storage.ChunkedSecretsConfig{
+		ChunkSize:      0,
+		MaxReadChunks:  0,
+		MaxWriteChunks: 0,
```

Comment:
You can just delete these fields from the struct initialization since they are the default values.